### ISSUE TITLE
Model: Income Minus Expenses By daily Cumulative

### ISSUE THREAD

**Issue by StrayaJake** [yet-another-tool/obsidian-findoc#24]
All the new models are great, use this on a daily

Wondering if its possible to use Model: Income Minus Expenses By Year Month, But instead it adds up Cumulative than takes the Cumulative, and for daily not yearly/monthly

IE
Income[1,2,3,4,5,6,7,8,9,10] - Expenses[1,2,3,4]
55 - 10

would be a single line similar to the other one, but instead daily

**Comment by studiowebux** [comment]
Hello !

Just to be sure that I correctly understood the requested model.

Let me know if it is what you need and I will push the model

---

It collects the data daily, sum per day (income) and subtract the sum per day (expenses), everything on a daily basis.

```
// Income: [
//     0, 30, 10,  0, 10, 10, 10, 40,
//    10, 10,  0, 10, 10, 10, 10, 10,
//    10, 10, 10, 10, 20, 80, 20, 20,
//    20, 20
//  ],
//  Expenses: [
//    5, 20, 10, 10,  0, 0, 130,  0,
//    0,  0, 10,  0, 20, 0,   0, 20,
//    0,  0,  0,  0,  0, 0,   0,  0,
//    0,  0,  0
//  ]


this will be the output: 
// results: [
// 	-5, 10, 0, -10, 10, 10, -120, 40, 10, 10, -10, 10, -10, 10, 10, -10, 10, 10,
// 	10, 10, 20, 80, 20, 20, 20, 20,
// ];
```

The dataset : 

```csv
Category,Subcategory,Value,TimeStamp,Extra
Income,Sale,10,2023-01-01,
Income,Sale,10,2023-01-02,
Income,Sale,10,2023-01-02,
Income,Sale,10,2023-01-02,
Income,Sale,10,2023-01-03,
Income,Sale,10,2023-01-06,
Income,Sale,10,2023-01-07,
Income,Sale,10,2023-01-08,
Income,Sale,10,2023-01-09,
Income,Sale,10,2023-01-09,
Income,Sale,10,2023-01-09,
Income,Sale,10,2023-01-09,
Income,Sale,10,2023-01-10,
Income,Sale,10,2023-01-11,
Income,Sale,10,2023-01-15,
Income,Sale,10,2023-01-16,
Income,Sale,10,2023-01-17,
Income,Sale,10,2023-01-19,
Income,Sale,10,2023-01-20,
Income,Sale,10,2023-01-21,
Income,Sale,10,2023-01-22,
Income,Sale,10,2023-01-24,
Income,Sale,10,2023-01-25,
Income,Sale,20,2023-02-05,
Income,Sale,20,2023-02-06,
Income,Sale,20,2023-02-06,
Income,Sale,20,2023-02-06,
Income,Sale,20,2023-02-06,
Income,Sale,20,2023-02-08,
Income,Sale,20,2023-02-12,
Income,Sale,20,2023-02-18,
Income,Sale,20,2023-02-24,
Expenses,Materials,5,2023-01-01,
Expenses,Materials,5,2023-01-02,
Expenses,Materials,5,2023-01-02,
Expenses,Materials,5,2023-01-02,
Expenses,Materials,5,2023-01-02,
Expenses,Materials,10,2023-01-03,
Expenses,Materials,10,2023-01-04,
Expenses,Materials,100,2023-01-08,
Expenses,Materials,10,2023-01-08,
Expenses,Materials,20,2023-01-08,
Expenses,Materials,10,2023-01-12,
Expenses,Materials,10,2023-01-16,
Expenses,Materials,10,2023-01-16,
Expenses,Materials,20,2023-01-20,
```

**Comment by StrayaJake** [comment]
Yeap that is exactly what I was thinking :)

**Comment by studiowebux** [comment]
Perfect !

I've released the version 0.6.1 with the new model:

You might need to update the `data.json` configuration file to add it manually: 
````json
,
    "incomeMinusExpensesByDaily": {
      "dataSource": "splitDailyDates",
      "categories": ["Income", "Expenses"],
      "output": "generateCumulativeDifference",
      "beginAtZero": true,
      "chartLabelType": "money",
      "dataSourceKey": "timestamp",
      "values": "Income, Expenses"
    }
````

Documentation: https://studiowebux.github.io/obsidian-plugins-docs/docs/findoc/models/incomeMinusExpensesByDaily

Let me know if there is anything

**Comment by studiowebux** [comment]
I released the version 0.6.2 that includes a button in the settings tab. 
it will load the default models (and removed the custom ones if any)
 this way you will get the model without manual interactions

**Comment by StrayaJake** [comment]
![image](https://github.com/yet-another-tool/obsidian-findoc/assets/1285708/cbbafe66-4583-441c-b725-1a49a56fc982)
I've updated to the latest, but it seems the Model itself output is empty,
have tried reinstalling, and fresh Vault and same issue is persistent

**Comment by studiowebux** [comment]
Morning
Oh ! you are right, I forgot one thing. I will fix ASAP

It should work even if it is not showing there, this field is only used to update the output function. If you load the defaults models it will update the configuration file correctly

thank you for catching and reporting it !

**Comment by studiowebux** [comment]
I've just released version 0.6.4 with few fixes, 
let me know if you have issues with it
thanks !

**Comment by StrayaJake** [comment]
All good here :)
